README.md

```diff
@@ -90,26 +91,82 @@ async fn main() {
 
 ```
 
-This pattern is useful if your child future needs to know about the cancel signal. This is
-highly useful in many situations where a child future needs to perform graceful termination.
+Contexts may also be chained by using the `with_parent` constructor in conjunction with
+RefContexts. Chaining a context means that the context will be cancelled if a parent context is
+cancelled. A `RefContext` is simple a wrapper type around an `Arc<Mutex<Context>>` with an
+identical API to `Context`. Here are a few examples to demonstrate how chainable contexts work:
+
+```rust
+use std::time::Duration;
+use tokio::time;
+use tokio::task;
+use tokio_context::context::RefContext;
+
+#[tokio::test]
+async fn cancelling_parent_ctx_cancels_child() {
+    // Note that we can't simply drop the handle here or the context will be cancelled.
+    let (parent_ctx, parent_handle) = RefContext::new();
+    let (mut ctx, _handle) = Context::with_parent(parent_ctx.clone(), None);
```

Comment:
So, it is actually possible because in this case a RefContext was passed down which is an Arc<Mutex>, so they are cloneable. I think if it was `parent_ctx.child()` that would imply it was already chained which I don't think would be possible off the parent context. It could be that it takes in a &RefContext and we clone it internally in the with_parent constructor.

Reply:
Done.